graphql_compiler/compiler/emit_sql.py

```
@@ -74,31 +77,249 @@ def _find_columns_used(sql_schema_info, ir):
+        """Generate output columns for innermost fold scope and add them to active SQLFoldObject."""
+        # find outputs for this fold in all_folded_outputs and add to self._outputs
+        if fold_scope_location.fold_path in all_folded_outputs:
+            for fold_output in all_folded_outputs[fold_scope_location.fold_path]:
+                # distinguish between folds with the same fold path but different query paths
+                if (fold_output.base_location, fold_output.fold_path) == (
+                        fold_scope_location.base_location, fold_scope_location.fold_path):
+                    if fold_output.field == '_x_count':
+                        raise NotImplementedError(u'_x_count not implemented in SQL')
+
+                    # force column to have explicit label as opposed to anon_label
+                    intermediate_fold_output_name = 'fold_output_' + fold_output.field
+                    # add array aggregated output column to self._outputs
+                    self._outputs.append(
+                        sqlalchemy.func.array_agg(
+                            self.output_vertex_alias.c[fold_output.field]
+                        ).label(intermediate_fold_output_name)
+                    )
+
+        # this is the unique identifier for the outer vertex used to join to the outer table
+        self._outputs.append(self.outer_vertex_alias.c[join_descriptor.from_column])
+
+        return sorted(self._outputs, key=lambda column: column.name, reverse=True)
+
+    def visit_output_vertex(self,
+                            output_alias,
+                            fold_scope_location,
+                            join_descriptor,
+                            all_folded_outputs):
+        """Update output columns when visiting the vertex containing output directives."""
+        self._output_vertex_alias = output_alias
+        self._outputs = self._get_fold_outputs(fold_scope_location,
```

we should probably raise an error here if a user tries to call `visit_output_vertex` more than once.